networkit/cpp/io/NetworkitBinaryWriter.cpp

```diff
@@ -17,12 +18,24 @@
 
 namespace NetworKit {
 
-NetworkitBinaryWriter::NetworkitBinaryWriter(uint64_t chunks, NetworkitBinaryWeights weightsType): chunks(chunks), weightsType(weightsType) {}
+NetworkitBinaryWriter::NetworkitBinaryWriter(uint64_t chunks, NetworkitBinaryWeights weightsType, bool preserveEdgeIndex): chunks(chunks), weightsType(weightsType), preserveEdgeIndex(preserveEdgeIndex) {}
 
 void NetworkitBinaryWriter::write(const Graph &G, const std::string &path) {
-
     std::ofstream outfile(path, std::ios::binary);
     Aux::enforceOpened(outfile);
+    writeData(outfile, G);
+    INFO("Written graph to ", path);
+}
+
+std::vector<uint8_t> NetworkitBinaryWriter::writeToBuffer(const Graph &G) {
+    std::stringstream outdata;
+    writeData(outdata, G);
+    std::string str = outdata.str();
+    return std::vector<uint8_t>(str.begin(), str.end());
+}
```

Comment:
This incurs a further copy. If we use a writable memory mapped file, the copy could be avoided. It would be fine to keep this code for now (while changing it later to a writable memory mapped file). On the other hand, using a writable file and replacing `outStream.write` by `memcpy` does not seem to be that difficult either. Do you want to change that now or keep it as it is for this PR?